cpp/cmake/External_tiff_files/test/rewrite_tag.c

```diff
@@ -0,0 +1,344 @@
+        fprintf (stderr, "Can't set SamplesPerPixel tag.\n");
+        goto failure;
+    }
+    if (!TIFFSetField(tif, TIFFTAG_ROWSPERSTRIP, rows_per_strip)) {
+        fprintf (stderr, "Can't set SamplesPerPixel tag.\n");
+        goto failure;
+    }
+    if (!TIFFSetField(tif, TIFFTAG_PLANARCONFIG, PLANARCONFIG_CONTIG)) {
+        fprintf (stderr, "Can't set PlanarConfiguration tag.\n");
+        goto failure;
+    }
+
+    for (i = 0; i < length; i++ )
+    {
+        if( !TIFFWriteEncodedStrip( tif, i, buf, 10 ) )
+        {
+            fprintf (stderr, "Can't write image data.\n");
+            goto failure;
+        }
+    }
+
+    TIFFClose(tif);
+
+    /* Test whether we can write tags. */
+
+    tif = TIFFOpen(filename, "r+");
+
+    if (!tif) {
+        fprintf (stderr, "Can't create test TIFF file %s.\n", filename);
+        return 1;
+    }
+
+    buf[3] = 17;
+    buf[6] = 12;
+
+    if( !TIFFWriteEncodedStrip( tif, 6, buf, 10 ) )
+    {
+        fprintf (stderr, "Can't write image data.\n");
+        goto failure;
+    }
+
+    TIFFClose(tif);
+
+    unlink(filename);
+
+    return 0;
+
+  failure:
+    /* Something goes wrong; close file and return unsuccessful status. */
+    TIFFClose(tif);
+    /*  unlink(filename); */
+
+    return 1;
+    
+}
+
+/************************************************************************/
+/*                            rewrite_test()                            */
+/************************************************************************/
+int rewrite_test( const char *filename, int length, int bigtiff, 
+                  uint64 base_value )
+
+{
+    TIFF		*tif;
+    int			i;
+    unsigned char	buf[10] = {5,6,7,8,9,10,11,12,13,14};
+    uint64		*rowoffset, *rowbytes;
+    uint64		*upd_rowoffset;
+    uint64		*upd_bytecount;
+
+    /* Test whether we can write tags. */
+    if( bigtiff )
+        tif = TIFFOpen(filename, "w8");
+    else
+        tif = TIFFOpen(filename, "w4");
+
+    if (!tif) {
+        fprintf (stderr, "Can't create test TIFF file %s.\n", filename);
+        return 1;
+    }
+
+    if (!TIFFSetField(tif, TIFFTAG_COMPRESSION, COMPRESSION_PACKBITS)) {
+        fprintf (stderr, "Can't set Compression tag.\n");
+        goto failure;
+    }
+    if (!TIFFSetField(tif, TIFFTAG_IMAGEWIDTH, width)) {
+        fprintf (stderr, "Can't set ImageWidth tag.\n");
+        goto failure;
+    }
+    if (!TIFFSetField(tif, TIFFTAG_IMAGELENGTH, length)) {
+        fprintf (stderr, "Can't set ImageLength tag.\n");
+        goto failure;
+    }
+    if (!TIFFSetField(tif, TIFFTAG_BITSPERSAMPLE, 8)) {
+        fprintf (stderr, "Can't set BitsPerSample tag.\n");
+        goto failure;
+    }
+    if (!TIFFSetField(tif, TIFFTAG_SAMPLESPERPIXEL, 1)) {
+        fprintf (stderr, "Can't set SamplesPerPixel tag.\n");
+        goto failure;
+    }
+    if (!TIFFSetField(tif, TIFFTAG_ROWSPERSTRIP, rows_per_strip)) {
+        fprintf (stderr, "Can't set SamplesPerPixel tag.\n");
```

Comment:
Sorry, I wasn't reading properly, you were referring to the incorrect string.  I'll push a fix to the upstream developers, but leave it untouched here until then since this is intended to be a verbatim copy.